aclk/aclk_util.c

```diff
@@ -281,30 +281,37 @@ const char *aclk_get_topic(enum aclk_topics topic)
  *
  * @param mode 0 - to reset the delay,
  *             1 - to advance a step and calculate sleep time [0 .. ACLK_MAX_BACKOFF_DELAY * 1000] ms
+ * @param min, max in seconds
  * @returns delay in ms
  *
  */
-#define ACLK_MAX_BACKOFF_DELAY 1024
-unsigned long int aclk_reconnect_delay(int mode)
-{
-    static int fail = -1;
-    unsigned long int delay;
 
-    if (!mode || fail == -1) {
+unsigned long int aclk_reconnect_delay(int mode, int base, unsigned long int min, unsigned long int max) {
+    static int attempt = -1;
+
+    if (!mode) {
+        attempt = -1;
+        return 0;
+    }
+
+    if (attempt < 0) {
         srandom(time(NULL));
-        fail = mode - 1;
+        attempt++;
```

Comment:
We can increase `attempt` in one statement before the `if` statement.

Reply:
simplified